Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -733,6 +745,10 @@ class Gx2Fitter {
     // and used for the final track
     std::size_t tipIndex = Acts::MultiTrajectoryTraits::kInvalid;
 
+    // Here we will store, how the ndf of the system. It automatically deduces
+    // if we want to fit e.g. q/p and adjusts itself later.
+    std::size_t ndfSystem = Acts::MultiTrajectoryTraits::kInvalid;
```

Comment:
```suggestion
    std::size_t ndfSystem = 0;
```

Reply:
I wanted to set this to ~`+inf`, because we need to update it later. If we somehow miss to update it, we reach the abort condition, that any number of measurements is not sufficient to properly fit the system.

If we would set it to `0`, we might try to fit with even a single measurement.

Reply:
But `Acts::MultiTrajectoryTraits::kInvalid` looks like a misuse. You can use `numerical_limits` if you want to use "inf"

Arguably the initial state is that we have no degrees of freedom this is why I suggested 0.

Reply:
But the system has inf, because we have no information about it yet. By describing it in our parametrisation, we bind it to a 6-parameter system.

Reply:
Hm well then it should be 6 IMO. But anyways `Acts::MultiTrajectoryTraits::kInvalid` is a misuse. You should not use it for this purpose because it is a `MultiTrajectoryTraits` and has nothing to do with numbers of degrees of freedom so either you use `std::numerical_limits::max`, if you think that number is appropriate, or you hardcode a number.

Reply:
Ups just saw you changed it already - don't mind me 😅